app/views/idv/phone_errors/failure.html.erb

```diff
@@ -7,7 +7,7 @@
       url: idv_gpo_path,
     },
     decorated_session.sp_name && {
-      url: return_to_sp_failure_to_proof_path,
+      url: return_to_sp_failure_to_proof_path(step: 'phone', location: 'failure'),
```

Comment:
Yeah, we can. I don't think it's strictly necessary that the location match the template name, but they're a convenient, existing reference. Dynamic could make it easier for future copy-paste though.

I'd also recently stumbled upon [`ParseControllerFromReferer`](https://github.com/18F/identity-idp/blob/main/app/services/parse_controller_from_referer.rb), and could imagine it or some equivalent logic to automatically pull this location value from the current controller method.

Reply:
> I'd also recently stumbled upon [`ParseControllerFromReferer`](https://github.com/18F/identity-idp/blob/main/app/services/parse_controller_from_referer.rb), and could imagine it or some equivalent logic to automatically pull this location value from the current controller method.

A simpler option may just be:

```suggestion
      url: return_to_sp_failure_to_proof_path(step: 'phone', location: request.params[:action]),
```

Reply:
Went with this in 4fab86e.

Reply:
One thing about `:action` is that sometimes we have controllers that render different templates from from the action they're in. May not be a common part of the FSM, but it does happen

I do not like using `Refererer` because then we're trusting the browser to send that header reliably, and they don't always, especially after redirects

Anyways I don't feel super strongly, if the `params[:action]` is the least amount of code that gets us the analytics we need, I'm happy with that.

Reply:
Discussed via Slack: Will plan to keep this as-is for now. We don't have any rendering of these templates outside the phone controller currently, and if we did, there's a good chance we'd _want_ the action of the invoking controller. The specs give us an extra bit of confidence as well.